scipy/special/_basic.pyi

```diff
@@ -0,0 +1,271 @@
+from __future__ import annotations
+
+from typing import (
+    Callable,
+    overload,
+    Literal,
+)
+
+from ._ufuncs import psi
+from numpy import sinc as sinc
+
+from numpy.typing import NDArray
+
+from scipy._lib._util import DecimalNumber, IntNumber
```

Comment:
`DecimalNumber` is a quite weird name for something that Python or NumPy `int` or `float` scalar. Let's try to fix that.

Reply:
Could you explain a little more? A more explanatory name is preferable, something like `_ScalarType`

Reply:
In this case I don't think you need `DecimalNumber` at all. As I wrote in another comment, `ArrayLike` will do instead of `DecimalNumber | NDArray`. 

In case you actually need scalars, there is probably a better way - see for example https://numpy.org/devdocs/reference/typing.html#number-precision. I can comment in more detail for a concrete case.